src/neml2/tensors/Scalar.cxx

```diff
@@ -42,4 +42,11 @@ operator*(const Scalar & a, const Scalar & b)
 {
   return torch::operator*(a, b);
 }
+
+Scalar
+abs(const Scalar & a)
+{
+  return Scalar(torch::abs(a), a.batch_dim());
+}
```

Comment:
hmm, so I guess the version in `BatchTensorBase.h` doesn't cover this case?

Reply:
I get a fun compiler error on some machines saying that even though it's not ambiguous, it actually is according to some obscure part of the C++ spec.